src/Koffee/MainController.fs

```diff
@@ -677,15 +677,20 @@ module Action =
                 | DeletedItem (item, permanent) ->
                     Error (CannotUndoDelete (permanent, item))
                     |> AsyncSeq.singleton
-            yield
+            let model =
                 { model with
                     RedoStack = action :: model.RedoStack
-                    Status = Some <| MainStatus.undoAction action model.PathFormat model.RepeatCount
+                    Status = Some <| MainStatus.undoAction action model.PathFormat (Some count) iter
                 }
+            yield model
```

Comment:
I think this `yield model` belongs in an `else` to the `if` below. Yielding in the middle of a repeated undo will only very briefly show a misleading status before the next iteration sets the status to another Busy status. The status only needs to be set in the else as well, because the next undo operation is just going to immediately set the status to a Busy status.

Reply:
Good observation!